ci/functional/test_main.sh

```diff
@@ -2,6 +2,33 @@
 
 set -eux
 
+report_junit() {
+    local name="$1"
+    local results="$2"
+    local nodes="$3"
+
+    clush -o '-i ci_key' -l root -w $nodes --rcopy "$2"
+
+    local num_reports="$(ls "$2".* | wc -l)"
```

Comment:
(lint) Declare and assign separately to avoid masking return values. [SC2155]
(lint) Use find instead of ls to better handle non-alphanumeric filenames. [SC2012]